components/script/dom/webidls/WindowOrWorkerGlobalScope.webidl

```diff
@@ -28,6 +28,11 @@ interface mixin WindowOrWorkerGlobalScope {
   Promise<ImageBitmap> createImageBitmap(ImageBitmapSource image, optional ImageBitmapOptions options = {});
   // Promise<ImageBitmap> createImageBitmap(
   //   ImageBitmapSource image, long sx, long sy, long sw, long sh, optional ImageBitmapOptions options);
+
+  // structured cloning
+  [Throws]
+  // any structuredClone(any value);
+  any structuredClone(any value, optional StructuredSerializeOptions options = {});
```

Comment:
codegen is not working for `StructuredSerializeOptions ` I am getting:

```
   File "servo/third_party/WebIDL/WebIDL.py", line 2800, in complete
      raise WebIDLError("Unresolved type '%s'." % self.name, [self.location])
  WebIDL.WebIDLError: error: Unresolved type '<unresolved scope>::StructuredSerializeOptions'., /servo/components/script/dom/bindings/codegen/../../webidls/WindowOrWorkerGlobalScope.webidl line 35:42
    any structuredClone(any value, optional StructuredSerializeOptions options = {});
    ```
    
    I am trying to figure it out

Reply:
I think the name changed, we have it at https://github.com/servo/servo/blob/3c271fb2989fa838473ab006397c7a9d8c3b4b21/components/script/dom/webidls/MessagePort.webidl#L21, so it would have to be defined as per https://html.spec.whatwg.org/multipage/web-messaging.html#structuredserializeoptions 

I think this can be done as part of this PR.

Reply:
Indeed it changed, I will update it.

Reply:
Renamed `PostMessageOptions` to `StructuredSerializeOptions`, ready for review